### DISCUSSION TITLE
Refactor Question

### DISCUSSION THREAD

**Discussion by Alexander-Frost** [mozilla-mobile/firefox-ios#10758] [Q&A] [answered]
What's the reason we have a webView in every tab?

We should be able to use a handful of webViews that share the same configuration across the entire app, for all tabs. We could hold onto the current page and `backForwardList` in each tab, and when a new tab is selected, we use the `SessionRestore.html` file to load up the page / tab. All the Private / Incognito tabs would use one WebView, and all normal tabs would use another.

What's the reason we don't do this?

**Comment by lmarceau** [comment, marked as answer]
Linking to discussion on issue here: https://github.com/mozilla-mobile/firefox-ios/issues/10766 - let's keep the conversation on the issue only so it's easier to find!
